### PR TITLE
 RJS-2757: Add Apple Privacy Manifest

### LINKED ISSUE #6547: Add privacy manifest to apple binaries

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #6547 

Looking at https://github.com/realm/realm-swift/pull/8455 for inspiration (aka stealing)

Related to https://github.com/react-native-community/discussions-and-proposals/discussions/776

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by elle-j — APPROVED** [review]
LGTM! 🙂

File: packages/realm/RealmJS.podspec [hunk 1]
```diff
@@ -43,6 +43,7 @@ Pod::Spec.new do |s|
 
   s.source_files           = 'react-native/ios/RealmReact/*.mm'
   s.public_header_files    = 'react-native/ios/RealmReact/*.h'
+  s.resource_bundles       = { 'RealmJS_privacy' => ['PrivacyInfo.xcprivacy'] }
```
Comment (kraenhansen) [inline packages/realm/RealmJS.podspec:46]: I was wondering if the `_privacy` suffix was needed and clicked through to this explanation: https://github.com/flutter/packages/pull/5846#issuecomment-1881788602

File: packages/realm/PrivacyInfo.xcprivacy [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kraenhansen) [inline packages/realm/PrivacyInfo.xcprivacy:1]: This filename would need to go into the "files" of the "package.json" too: https://github.com/realm/realm-js/blob/ec7a050c95b50672135becbb9fd8e38108b300ce/packages/realm/package.json#L52-L74

File: packages/realm/RealmJS.podspec (hunk 1 above)
Comment (kneth) [reply packages/realm/RealmJS.podspec:46]: Let's remove `_privacy`

**Pull request merged**